cookbook/examples/workflows/content_creator_workflow/workflow.py

```diff
@@ -0,0 +1,186 @@
+        model=OpenAIChat(id="gpt-4o"),
+        description=f"{agents_config['twitter_thread_planner']['role']} - {agents_config['twitter_thread_planner']['goal']}",
+        instructions=[
+            f"{agents_config['twitter_thread_planner']['backstory']}",
+            tasks_config["create_twitter_thread_plan"]["description"],
+        ],
+        response_model=Thread,  # Expects response to follow the Thread Pydantic model
+    )
+
+    # LinkedIn Post Planner: Converts blog content into a structured LinkedIn post, optimized for a professional
+    # audience with relevant hashtags.
+    linkedin_post_planner: Agent = Agent(
+        model=OpenAIChat(id="gpt-4o"),
+        description=f"{agents_config['linkedin_post_planner']['role']} - {agents_config['linkedin_post_planner']['goal']}",
+        instructions=[
+            f"{agents_config['linkedin_post_planner']['backstory']}",
+            tasks_config["create_linkedin_post_plan"]["description"],
+        ],
+        response_model=LinkedInPost,  # Expects response to follow the LinkedInPost Pydantic model
+    )
+
+    def scrape_blog_post(self, blog_post_url: str, use_cache: bool = True):
+        if use_cache and blog_post_url in self.session_state:
+            logger.info(f"Using cache for blog post: {blog_post_url}")
+            return self.session_state[blog_post_url]
+        else:
+            response: RunResponse = self.blog_analyzer.run(blog_post_url)
+            if isinstance(response.content, BlogAnalyzer):
+                result = response.content
+                logger.info(f"Blog title: {result.title}")
+                self.session_state[blog_post_url] = result.blog_content_markdown
+                return result.blog_content_markdown
+            else:
+                raise ValueError("Unexpected content type received from blog analyzer.")
+
+    def generate_plan(self, blog_content: str, post_type: PostType) -> dict:
+        plan_response: RunResponse = RunResponse(content=None)
+        if post_type == PostType.TWITTER:
+            logger.info(f"Generating post plan for {post_type}")
+            plan_response = self.twitter_thread_planner.run(blog_content)
+        elif post_type == PostType.LINKEDIN:
+            logger.info(f"Generating post plan for {post_type}")
+            plan_response = self.linkedin_post_planner.run(blog_content)
+        else:
+            raise ValueError(f"Unsupported post type: {post_type}")
+
+        if plan_response.content_type == "Thread" or plan_response.content_type == LinkedInPost:
+            return plan_response.content
+        elif plan_response.content_type == "application/json":
+            data = json.loads(plan_response.content)
+            if post_type == PostType.TWITTER:
+                return Thread(**data)
+            else:
+                return LinkedInPost(**data)
+        else:
+            raise ValueError("Unexpected content type received from planner.")
```

Comment:
This logic will need a revisit as it's not clean. Look at line 131.
Also the return type is dict but you are returning 3 completely different types at line 132, 136, and 138